core/src/main/java/hudson/slaves/OfflineCause.java

```diff
@@ -60,11 +63,12 @@
          */
         protected SimpleOfflineCause(Localizable description) {
             this.description = description;
+            this.timestamp = new Date();
         }
 
         @Exported(name="description") @Override
         public String toString() {
-            return description.toString();
+            return "[" + timestamp.toString() + "] " + description.toString();
```

Comment:
I was never fond of using `toString()` to display `OfflineCause`. I would like to see this moved to jelly/groovy and displayed via `i:formatDate`.